tests/integration/security/mtlscert_pluginca_securenaming/mtlscert_pluginca_securenaming_test.go

```diff
@@ -0,0 +1,246 @@
+// limitations under the License.
+
+package mtlscertplugincasecurenaming
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/cert"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+const (
+	// The length of the example certificate chain.
+	exampleCertChainLength = 3
+
+	defaultIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/default"
+`
+	correctIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/b"
+`
+	nonExistIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "I-do-not-exist"
+`
+	identityListDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/a"
+      - "spiffe://cluster.local/ns/NS/sa/b"
+      - "spiffe://cluster.local/ns/default/sa/default"
+      - "I-do-not-exist"
+`
+)
+
+// TestMTLSCertPluginCASecureNaming verifies:
+// - The certificate issued by CA to the sidecar is as expected and that strict mTLS works as expected.
+// - The plugin CA certs are correctly used in workload mTLS.
+// - The CA certificate in the configmap of each namespace is as expected, which
+//   is used for data plane to control plane TLS authentication.
+// - Secure naming information is respected in the mTLS handshake.
+func TestMTLSCertPluginCASecureNaming(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+			testNamespace := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "test-mtls",
```

Comment:
this is still test-mtls

Reply:
I didn't change it because it's the only namespace in the testing cluster. But I can do that.